### PR TITLE
feat(1769): [3] Change data structure for queue-service cooperation. BREAKING CHANGE: screwdriver-api no longer works.

### PR DESCRIPTION
**Pull request by yk634**
## Context

Change the data structure so that [getChangedFiles](https://github.com/screwdriver-cd/scm-github/blob/3eb11b795ea050f02e49f28994a1424165869fa4/index.js#L1318) can be executed from the endpoint `/processHooks` added in https://github.com/screwdriver-cd/screwdriver/pull/2622.

## Objective

- parseHook
  - Add the three items needed to do [getChangedFiles](https://github.com/screwdriver-cd/scm-github/blob/3eb11b795ea050f02e49f28994a1424165869fa4/index.js#L1318) to the return value.
- getChangedFiles
  - Change to retrieve the changed file with the information received from [parseHook](https://github.com/screwdriver-cd/scm-github/blob/3eb11b795ea050f02e49f28994a1424165869fa4/index.js#L1368).

This is a breaking change, because the arguments to getChangedFiles will be renamed, and the caller of that function will need to be modified as well.
That modification is done at https://github.com/screwdriver-cd/screwdriver/pull/2622.

## References

Because of the changes in the data structure, the unit test may fail until the [changes in data-schema](https://github.com/screwdriver-cd/data-schema/pull/471) are published.

[data-schema PR](https://github.com/screwdriver-cd/data-schema/pull/471)
[scm-base PR](https://github.com/screwdriver-cd/scm-base/pull/90)
[scm-gitlab PR](https://github.com/screwdriver-cd/scm-gitlab/pull/49)
[screwdriver PR](https://github.com/screwdriver-cd/screwdriver/pull/2622)

https://github.com/screwdriver-cd/screwdriver/issues/1769#issuecomment-934107626

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -13,11 +13,13 @@ const testPayloadPushTag = require('./data/github.push.tag.json');
 const testPayloadRelease = require('./data/github.release.json');
 const testPayloadReleaseBadAction = require('./data/github.release.badAction.json');
 const testPayloadTag = require('./data/github.tag.json');
-const testPayloadPushBadHead = require('./data/github.push.badHead.json');
```
Comment (tk3fftk) [inline test/index.test.js:16]: Can we also remove `./data/github.push.badHead.json` file?

File: test/data/webhookConfig.pull_request.opened.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "ref": "pull/1/merge",
+  "prTitle": "Update the README with new information",
+  "sha": "0d1a26e67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c",
+  "prSource": "branch",
+  "type": "pr",
+  "username": "baxterthehacker2",
+  "hookId": "3c77bf80-9a2f-11e6-80d6-72f7fe03ea29",
+  "scmContext": "github:github.com",
+  "action": "opened"
+}
```
Comment (tk3fftk) [inline test/data/webhookConfig.pull_request.opened.json:15]: 👀

File: index.js [hunk 3]
```diff
@@ -1308,14 +1308,14 @@ class GithubScm extends Scm {
...
-     * @return {Promise}                   Resolves to an array of filenames of the changed files
+     * @param  {String}   config.type               Can be 'pr' or 'repo'
+     * @param  {Object}   [config.webhookConfig]    The webhook payload received from the SCM service.
+     * @param  {String}   config.token              Service token to authenticate with Github
+     * @param  {String}   [config.scmUri]           The scmUri to get PR info of
+     * @param  {Integer}  [config.prNum]            The PR number
+     * @return {Promise}                            Resolves to an array of filenames of the changed files
      */
-    async _getChangedFiles({ type, payload, token, scmUri, prNum }) {
+    async _getChangedFiles({ type, webhookConfig, token, scmUri, prNum }) {
```
Comment (wahapo) [inline index.js:1318]: We also need to fix scm-base.
https://github.com/screwdriver-cd/scm-base/blob/f900e1579391dfd4cdd5ac0f465f56a284265e53/index.js#L173

**Review by wahapo — COMMENTED** [review]
Does this need `BREAKING CHANGE: MESSAGE`?
https://docs.screwdriver.cd/about/contributing/

File: test/index.test.js (hunk 1 above)
Comment (yk634) [reply test/index.test.js:16]: I removed it.

File: test/data/webhookConfig.pull_request.opened.json (hunk 2 above)
Comment (yk634) [reply test/data/webhookConfig.pull_request.opened.json]: I added new line to the files.

File: index.js (hunk 3 above)
Comment (yk634) [reply index.js:1318]: I fixed it at https://github.com/screwdriver-cd/scm-base/pull/90.

**Review by wahapo — APPROVED** [review]

**Review by y-oksaku — APPROVED** [review]

**Review by itleigns — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Review by kumada626 — APPROVED** [review]

**Review by ibu1224 — APPROVED** [review]

**Pull request merged**